### PR TITLE
feat: add snappy compression for ocf

### PR DESCRIPTION
**Pull request by macasado86**
This PR adds functionality to compress Object Container Files with Snappy (snappyer dependency has been added). 

https://avro.apache.org/docs/current/spec.html#snappy

### CODE REVIEW

File: src/avro_ocf.erl [hunk 1]
```diff
@@ -253,10 +255,23 @@ decode_block(_Lkup, _Type, _Codec, <<>>, 0, Acc, _Options) -> Acc;
 decode_block(Lkup, Type, deflate, Bin, Count, Acc, Options) ->
   Decompressed = zlib:unzip(Bin),
   decode_block(Lkup, Type, null, Decompressed, Count, Acc, Options);
+decode_block(Lkup, Type, snappy, Bin, Count, Acc, Options) ->
+  Size = byte_size(Bin),
+  <<Compressed:(Size-4)/binary, Checksum:4/binary>> = Bin,
+  {ok, Decompressed} = snappyer:decompress(Compressed),
+  case is_invalid_checksum(Decompressed, Checksum) of
+    false -> Decompressed;
```
Comment (mikpe) [inline src/avro_ocf.erl:263]: I'm not a big fan of double-negatives. Can you change the predicate to the positive `is_valid_checksum/2` and put the `true` case first?

File: src/avro_ocf.erl [hunk 2]
```diff
@@ -253,10 +255,23 @@ decode_block(_Lkup, _Type, _Codec, <<>>, 0, Acc, _Options) -> Acc;
...
+    true -> erlang:error({invalid_checksum, Decompressed})
+  end,
+  decode_block(Lkup, Type, null, Decompressed, Count, Acc, Options);
 decode_block(Lkup, Type, null, Bin, Count, Acc, Options) ->
   {Obj, Tail} = decode_stream(Lkup, Type, Bin, Options),
   decode_block(Lkup, Type, null, Tail, Count - 1, [Obj | Acc], Options).
 
+- spec is_invalid_checksum(binary(), binary()) -> boolean().
+is_invalid_checksum(Decompressed, Checksum) ->
+  binary:encode_unsigned(erlang:crc32(Decompressed)) /= Checksum.
```
Comment (mikpe) [inline src/avro_ocf.erl:273]: nit: as this doesn't involve float/integer conversions I'd prefer exact inequality `=/=` here

**Review by mikpe — COMMENTED** [review]
Looks good modulo one or two minor coding-style nits. I'll take a closer look on Monday.

File: src/avro_ocf.erl (hunk 1 above)
Comment (macasado86) [reply src/avro_ocf.erl]: Changed! I have renamed the function to `is_valid_checksum`

File: src/avro_ocf.erl (hunk 2 above)
Comment (macasado86) [reply src/avro_ocf.erl]: Changed to `=:=` since I have renamed the function to `is_valid_checksum`

**Review by jesperes — APPROVED** [review]

**Review by mikpe — APPROVED** [review]

**Comment by mikpe** [comment]
Tagged as 2.9.6

**Comment by macasado86** [comment]
Thank you very much!

**Pull request merged**